### DISCUSSION TITLE
MenuItem KeyGesture and HotkeyManager

### DISCUSSION THREAD

**Discussion by tkefauver** [AvaloniaUI/Avalonia#12759] [Q&A]
I use `KeyGesture`s in `ContextMenu` `MenuItem`s  to only **show** the shortcuts. My app listens for global hotkeys using [SharpHook](https://github.com/TolikPylypchuk/SharpHook) and also handles any application shortcuts too so I never use the `HotkeyManager`.

I've had a long standing intermittent bug where I'm not receiving releases for some keys (totally random keys, input or mod both happen) and am wondering if maybe the `HotkeyManager` could be internally doing something to suppress the releases so my global handler doesn't receive them.

Is there a way to disable the `HotkeyManager`? 

Or am I wrong in thinking its even listening? I never use `HotKeyManager.SetHotKey()` anywhere and create the `MenuItem`s in code but not sure.

**Comment by timunie** [comment]
Not sure about what's going on here. Maybe you want to file a minimal sample, so someone experienced could probably give you a hint.

**Comment by tkefauver** [comment]
Not code specific, I was just giving context about the bug which isn't a problem with Avalonia. 
The questions were:

> Is there a way to disable the HotkeyManager?

And

> Or am I wrong in thinking its even listening? I never use HotKeyManager.SetHotKey() anywhere and create the MenuItems in code but not sure.

**Comment by TolikPylypchuk** [comment]
Hi! I'm the author of SharpHook. This issue is not related to Avalonia and is already [fixed](https://github.com/TolikPylypchuk/SharpHook/issues/61) in SharpHook. Just FYI.
